contracts/crowdfund/ContributionRouter.sol

```diff
@@ -52,11 +54,28 @@ contract ContributionRouter {
     ///         and keeps the fee amount. The target contract is expected to
     ///         be appended to the calldata.
     fallback() external payable {
-        uint256 feeAmount = feePerContribution;
+        uint256 feeAmount = feePerMint;
         address target;
         assembly {
             target := shr(96, calldataload(sub(calldatasize(), 0x14)))
         }
+        if (msg.sig == InitialETHCrowdfund.batchContributeFor.selector) {
+            uint256 numOfMints;
+            assembly {
+                // 228 is the offset of the length of `tokenIds` in the
+                // calldata.
+                numOfMints := calldataload(228)
+            }
+            feeAmount *= numOfMints;
+        } else if (msg.sig == Crowdfund.batchContributeFor.selector) {
```

Comment:
Given that we don't plan on using this functionality, I don’t think it’s worth adding.